src/components/LoginPage.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+import { AuthRequest } from "../models/request/AuthRequest";
+import { setCredentials } from "../redux/authSlice";
+import { useLoginUserMutation } from "../services/AuthService";
+import Button from "./UI/Button";
+import CreateOrLoginComponent from "./UI/CreateOrLoginComponent";
+import Form from "./UI/Form";
+import Input from "./UI/Input";
+
+const Container = styled.div`
+  margin-top: 5rem;
+  max-width: 20rem;
+  width: 100%;
+  height: auto;
+  margin: 7rem auto;
+`;
+
+const LoginPage = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [emailError, setEmailError] = useState("Email cant't be empty");
```

Comment:
there is too much logic about that. Let's imagine we have 5 or 6 fields to fill. then you'll use a lot of `useState` - one for each field, one for possible issue with field; 
+I'm really not sure whats goin on with `dirty` fields eg `dirtyPassword` :)

possible solutions:
keep using useState for every field +one common for errors. eg [hasError, setHasError] and then one common validation function that will find and notify user about errors. 

ps. if you use multiple useState with a different variations, it's useful to use react reducer:
https://reactjs.org/docs/hooks-reference.html#usereducer